include/vsbase.hpp

```diff
@@ -161,25 +167,30 @@ class ThermoPropBase {
     rsGrid.push_back(0.0);
     const double rsMax = rs + drs;
     while(!numUtil::equalTol(rsGrid.back(), rsMax)){
-      rsGrid. push_back(rsGrid.back() + drs);
+      rsGrid.push_back(rsGrid.back() + drs);
     }
+    // Resize the free parameter vector
+    const size_t nrs = rsGrid.size();
+    alpha.resize(nrs);
     // Initialize the free energy integrand
     fxcIntegrand.resize(NPOINTS);
-    const size_t nrs = rsGrid.size();
     for (auto& f : fxcIntegrand) {
       f.resize(nrs);
       vecUtil::fill(f, numUtil::Inf);
     }
-    // Fill the free energy integrand if passed in input
+    // Fill the free energy integrand and the free parameter if passed in input
     const auto& fxciData = in.getFreeEnergyIntegrand();
     if (!fxciData.grid.empty()) {
       for (const auto& theta : {Idx::THETA_DOWN, Idx::THETA, Idx::THETA_UP}) {
-	const Interpolator1D itp(fxciData.grid, fxciData.integrand[theta]);
-	const double rsMaxi = fxciData.grid.back();
-	for (size_t i = 0; i < nrs; ++i) {
-	  const double& rs = rsGrid[i];
-	  if (rs <= rsMaxi) { fxcIntegrand[theta][i] = itp.eval(rs); }
-	}
+        const double rsMaxi = fxciData.grid.back();
+        const Interpolator1D itp(fxciData.grid, fxciData.integrand[theta]);
+        for (size_t i = 0; i < nrs; ++i) {
+          const double& rs = rsGrid[i];
+          if (rs <= rsMaxi) { fxcIntegrand[theta][i] = itp.eval(rs); }
+          if (theta == Idx::THETA && rs <= rsMaxi) {
+            alpha[i] = fxciData.alphaData[i];

```

Comment:
There is no need to check again rs if you put this inside the same if block as the one you use to fill the free energy integrand. Actually, you could also avoid having to check theta if you fill alpha and the free energy integrand separately. I leave the choice to you